File: Codes/Programmers/기능개발.py
from math import ceil


def solution(progresses, speeds):
    answer = []
    remain = [ceil((100 - p) / s) for p, s in zip(progresses, speeds)]
    now = 0
    for time in remain:
        if time > now:
            answer.append(1)
            now = time
        else:
            answer[-1] += 1

    return answer

File: Codes/Programmers/기능개발.js
function solution(progresses, speeds) {
  const remainTime = progresses.map((p, idx) =>
    Math.ceil((100 - p) / speeds[idx])
  );

  var answer = [];
  let now = 0;
  remainTime.forEach((time) => {
    if (time > now) {
      answer.push(1);
      now = time;
    } else {
      answer[answer.length - 1]++;
    }
  });
  return answer;
}
